compose/ui/ui/src/webCommonW3C/kotlin/androidx/compose/ui/window/ComposeWindow.web.kt

```diff
@@ -211,20 +211,8 @@ internal class ComposeWindow(
             override fun getNewGeometryForBackingInput(rect: Rect): DpRect {
                 val viewportRect = canvas.getBoundingClientRect()
                 val dpRect = rect.toDpRect(density)
-                var left = viewportRect.left.toFloat() + dpRect.left.value
-                var top = viewportRect.top.toFloat() + dpRect.top.value
-
-                if (left < viewportRect.left) {
-                    left = viewportRect.left.toFloat()
-                } else if (left + dpRect.width.value > viewportRect.right) {
-                    left = viewportRect.right.toFloat() - dpRect.width.value
```

Comment:
I believe this logic is still needed at least until we have a rectangle which corrensponds to the bounds of a textfield. 

`val left = viewportRect.left.toFloat() + dpRect.left.value` - might lead to a situation when the backing input bounds would exceed the canvas.

Reply:
@eymar let me apply exactly the same CSS-backed approach to the left positiononing as well then, it's pretty straightforward!